cmd/ttn-lw-migrate/main.go

```diff
@@ -21,6 +21,8 @@ import (
 	_ "go.thethings.network/lorawan-stack-migrate/pkg/source/ttnv2"      // TTNv2 source
 	_ "go.thethings.network/lorawan-stack-migrate/pkg/source/ttnv3"      // TTS source
 
+	_ "go.thethings.network/lorawan-stack-migrate/pkg/arg" // Init after initializing sources
```

Comment:
So we are doing this, such that `arg.init` runs before the `pflag` initialization ?

I am not fully against this idea, but keep in mind that you can edit the arguments of a `cobra.Command` using [`SetArgs`](https://pkg.go.dev/github.com/spf13/cobra#Command.SetArgs), perhaps in the `init` of `cmd/root`.

This in turns allows you basically intercept the parameters between `os.Args` and the actual usage, thus avoiding the clever-but-complicated solution.

Reply:
Using [`SetArgs`](https://pkg.go.dev/github.com/spf13/cobra#Command.SetArgs) instead of removing the field from `os.Args` seems like a great idea. The problem is that `application` command's `init` function is always executed first. The argument has to be interpreted before [`source.FlagSet`](https://github.com/TheThingsNetwork/lorawan-stack-migrate/blob/master/pkg/source/source.go#L79) is called for it to work.

Reply:
I see - `ApplicationFlagSet` and `DevicesFlagSet` depend on the active source being set in order to have the intended behavior. Instead of depending on `init` order, let's just move the initialization after the active source has been set.

Let me know what you think of f173c6c6367566ad30050998db8c2c2ef0486aef.

Reply:
It looks great